control_plane/src/attachment_service.rs

```diff
@@ -220,9 +241,213 @@ impl AttachmentService {
             .expect("non-Unicode path")
     }
 
+    /// PIDFile for the postgres instance used to store attachment service state
+    fn postgres_pid_file(&self) -> Utf8PathBuf {
+        Utf8PathBuf::from_path_buf(
+            self.env
+                .base_data_dir
+                .join("attachment_service_postgres.pid"),
+        )
+        .expect("non-Unicode path")
+    }
+
+    /// In order to access database migrations, we need to find the Neon source tree
+    async fn find_source_root(&self) -> anyhow::Result<Utf8PathBuf> {
+        // We assume that either prd or our binary is in the source tree. The former is usually
+        // true for automated test runners, the latter is usually true for developer workstations. Often
+        // both are true, which is fine.
+        let candidate_start_points: Vec<Utf8PathBuf> = vec![
+            // Current working directory
+            Utf8PathBuf::from_path_buf(std::env::current_dir()?).unwrap(),
+            // Directory containing the binary we're running inside
+            Utf8PathBuf::from_path_buf(env::current_exe()?.parent().unwrap().to_owned()).unwrap(),
+            // Source dir on github action workers
+            Utf8PathBuf::from_str("/__w/neon").unwrap(),
```

Comment:
I wonder if we should add reading from an env var as well to allow customization without having to change the source code.

Reply:
I've just removed this -- I think in practice the pwd or binary dir is enough, as github actions run with the pwd in the source tree.  If it isn't we'll find out soon!.